src/Go.CodeAnalysis/Parser/ImportNode.cs

```diff
@@ -6,29 +6,105 @@
     {
-        public ImportNode(SnapshotSegment extent, SnapshotSegment importExtent) : base(extent, ImmutableArray<ParseNode>.Empty)
+        public ImportNode(
+            SnapshotSegment extent,
+            ImmutableArray<ImportDeclarationNode> importDeclarations)
+            : base(extent, importDeclarations.CastArray<ParseNode>())
         {
-            this.ImportExtent = importExtent;
+            this.ImportDeclarations = importDeclarations;
         }
 
-        public SnapshotSegment ImportExtent { get; }
+        /// <summary>
+        /// All imported packages under this import line or block.
+        /// </summary>
+        public ImmutableArray<ImportDeclarationNode> ImportDeclarations { get; }
 
         public static bool TryParse(Lexer lexer, IList<Error> errors, out ImportNode parseNode)
         {
             var start = lexer.CurrentLexeme.Extent.Start;
+            parseNode = null;
 
-            if (!lexer.IsCorrectLexemeKeywordOrReportError(Keywords.Import, errors) ||
-                !lexer.TryAdvanceLexemeOrReportError(errors) ||
-                !lexer.IsCorrectLexemeTypeOrReportError(LexemeType.String, errors))
+            // Parse "import" keyword.
+            if (!lexer.IsCorrectLexemeKeywordOrReportError(Keywords.Import, errors)
+                || !lexer.TryAdvanceLexemeOrReportError(errors))
             {
-                parseNode = null;
                 return false;
             }
 
-            var extent = new SnapshotSegment(lexer.Snapshot, start, lexer.CurrentLexeme.Extent.End - start);
-            parseNode = new ImportNode(extent, lexer.CurrentLexeme.Extent);
+            // Single line import (string immediately following import keyword).
+            if (!lexer.CurrentLexeme.Extent.Equals("("))
+            {
+                ImportDeclarationNode singleLineDeclaration = null;
+                if (ImportDeclarationNode.TryParse(lexer, errors, out singleLineDeclaration))
+                {
+                    var singleLineImportExtent = new SnapshotSegment(lexer.Snapshot, start, lexer.CurrentLexeme.Extent.End - start);
+                    parseNode = new ImportNode(
+                        singleLineImportExtent,
+                        ImmutableArray.Create(singleLineDeclaration));
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            // Block import.
+
+            // Advance over opening parenthesis.
+            if (!lexer.TryAdvanceLexemeOrReportError(errors))
+            {
+                return false;
+            }
+
+            var declarationArrayBuilder = ImmutableArray.CreateBuilder<ImportDeclarationNode>();
+            bool isEndOfImportBlock = false;
+
+            // Parse each import declaration.
+            while(!isEndOfImportBlock)
+            {
+                // TODO: solve the issue that multi-line block contains inserted semicolun.
```

Comment:
Lexer semicolon insertion behavior is based on https://golang.org/ref/spec#Semicolons. Any deviations from that spec need to be fixed in the lexer. Otherwise, this might just be an expectation of the grammar.

Reply:
Ah, here we go: the delimiter is semi-colon, not comma: https://golang.org/ref/spec#Import_declarations

The lexer automatically adds semi-colons at end line.

Reply:
Yes I understand that rule. This TODO is for myself to add semicolon handling here.